bodhi/client/__init__.py

```diff
@@ -175,9 +176,21 @@ def new(user, password, url, **kwargs):
             traceback.print_exc()
 
 
+def _validate_edit_update(ctx, param, value):
+    """
+    Callback used by click to validate the update argument given to the updates edit command.
+    the update argument can only be update id or update title
+    """
+    if re.search(bindings.UPDATE_ID_RE, value)\
+       or re.search(bindings.UPDATE_TITLE_RE, value):
+        return value
+    else:
+        raise click.BadParameter("Please provide an Update ID or an Update Title")
+
+
 @updates.command()
 @add_options(new_edit_options)
-@click.argument('updateid')
+@click.argument('update', callback=_validate_edit_update)
 @url_option
 def edit(user, password, url, **kwargs):
     """
```

Comment:
You could mention here that ```UPDATE``` can be the update alias (FEDORA-blah) or the title. I think that would cause ```--help``` to display that info to the user. Optional.